### DISCUSSION TITLE
Need for guidance : creating a SWIG wrapper of a C++ project which includes spdlog

### DISCUSSION THREAD

**Discussion by jsn-inrs** [gabime/spdlog#2808] [Q&A]
Hi everybody, I've recently discovered spdlog (because the OpenSim musculoskeletel simulation engine uses it). I need to create a SWIG-Python wrapper based on OpenSim features, so it must also include spdlog instructions. Fot the moment, I'm starting with a minimum C++ class, see https://github.com/jsn-inrs/GravForcesProcessor for the entire project.
 
Unfortunately, generating the swig wrapping fails with the following errors:
1>Swig compile GravForcesProcessor.i for python
SwigGravityForcesProcessor\GravForcesProcessor.h(15): error : Unable to find 'spdlog\spdlog.h'
1>SwigGravityForcesProcessor\GravForcesProcessor.i(20): error : Unable to find 'spdlog\spdlog.h'
1>SwigGravityForcesProcessor\GravForcesProcessor.i(23): error : Unable to find 'spdlog\common.h'
1>SwigGravityForcesProcessor\GravForcesProcessor.i(24): error : Unable to find 'spdlog\logger.h'
1>SwigGravityForcesProcessor\GravForcesProcessor.i(25): error : Unable to find 'spdlog\sinks\stdout_color_sinks.h'
1>SwigGravityForcesProcessor\GravForcesProcessor.i(26): error : Unable to find 'spdlog\fmt\ostr.h'

Could somebody help me fix these errors ? I'm not sure what files are to be included and how (using CMake).

Thank you very much for you valuable help !

**Comment by tt4g** [comment]
I don't know, I have never used SWIG.
But it seems that include paths added by `include_directories()` do not affect SWIG.

See: https://gitlab.kitware.com/cmake/cmake/-/issues/18003
See also: https://stackoverflow.com/questions/61132555/swig-with-cmake-not-able-to-find-header-includes

Enable `SWIG_USE_TARGET_INCLUDE_DIRECTORIES` as suggested in the comments.

```cmake
set_target_properties(swigGravForcesProcessor
  PROPERTIES
    SWIG_USE_TARGET_INCLUDE_DIRECTORIES TRUE
)
```
